### PR TITLE
Modal work.

### PR DESCRIPTION
**Pull request by bnhovde**
Had 30 minutes to kill.
- Added escape key functionality.
- Made the overlay clickable to close

I've also added some data-attributes to the body. I've found this really useful in other projects where I want to quickly see what JS components a page has by just looking at the body attributes.

As always, happy to review/change any approach here.

### CODE REVIEW

**Comment by chris-pearce** [comment]
LGTM.

> I've also added some data-attributes to the body. I've found this really useful in other projects where I want to quickly see what JS components a page has by just looking at the body attributes.

That's a good idea, I like it :+1:.

I was thinking—and this isn't a TODO for this PR as we're all working on this component (well @bnhovde is :smile:)—that perhaps this component should be called **Dialog** as from what I understand **Modal** is a type of dialog behaviour, from MDN:

> Dialogs can be modal or non-modal. When a modal dialog appears on the screen, it's not possible to interact with any page content outside the dialog. In other words, the main application UI or page content is considered to be temporarily disabled as long as the modal dialog is showing. For non-modal dialogs it is still possible to interact with content outside of the dialog while the dialog is showing. Note that for non-modal dialogs there will have to be a global keyboard shortcut that allows focus to be moved between opened dialogs and the main page. For more information see the Managing Modal and Non Modal Dialogs guide. 
> —https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_dialog_role

So we could have a new addition to the `userOptions` object: `isModal` that'll disable all the ways to close a dialog except when an action within the dialog is actioned e.g. 2 buttons: **OK** and **Cancel**?

There is also an **Alert Dialog** type: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alertdialog_role which are always **modal**.

Out of interest I thought I'd see where the `dialog` element support is at, it's dismal: http://caniuse.com/#search=dialog%20element. More info from MDN: https://developer.mozilla.org/en/docs/Web/HTML/Element/dialog.

**Comment by bnhovde** [comment]
Yeah Dialog makes sense then. I've always thought modal was a term for overlays, but looking at what you've found here it's not and **Dialog** is the right name for the component.

RE: Like the idea of the `isModal` (or maybe `dialogType` to encompass the alert variant and others) config option. This should be a quick job. I'll have a go at that tomorrow.

Never knew there was a `dialog` element. Maybe support is so poor because people don't know it exists, create their own solutions and don't demand it implemented.

**Comment by bnhovde** [comment]
Finally got around to this.
- Renamed all references of modal to dialog
- Added optional `isModal` config option that determines if escape closes modal or not
- Updated demo
- Started writing tests for the module but hit a wall. Will pick this up later.

Other stuff:
- Deleted dead .js files (library etc)

File: src/components/dialog/dialog.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    var DOM,
+        state = {};
+        
+    const defaults = {
+        dialog:         '.js-dialog',
+        openBtn:        '.js-dialog-btn',
+        closeBtn:       '.js-dialog-close-btn',
+        backdrop:       '.js-dialog-backdrop',
+        isModal:        false,
+        showBackdrop:   true
```
Comment (chris-pearce) [inline src/components/dialog/dialog.js:18]: One thing me and @davislurve  are doing at work is starting all `boolean` variables with _is_ or _has_ so this one would be something like `hasBackdrop`. How do you feel about that naming convention?

File: src/components/dialog/dialog.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
+/**
+ * @function UIDialog
+ * @version 0.0.2
+ * @desc display a Dialog
+ */
+
+export default function UIDialog(userOptions) {
+
+    var DOM,
+        state = {};
```
Comment (chris-pearce) [inline src/components/dialog/dialog.js:10]: Just curious why this wouldn't be a `let`?

File: src/components/dialog/dialog.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        // Save all DOM queries for future use
+        DOM = {
+            'page': document.querySelectorAll('body')[0],
+            'dialog': document.querySelectorAll(settings.dialog)[0],
+            'openBtn': document.querySelectorAll(settings.openBtn)[0],
+            'closeBtn': document.querySelectorAll(settings.closeBtn)[0],
+            'backdrop': document.querySelectorAll(settings.backdrop)[0]
+        };
+        
+        // Check if gallery exists, return if not
```
Comment (chris-pearce) [inline src/components/dialog/dialog.js:41]: Replace _gallery_ with _dialog_ :+1:.

File: src/components/dialog/dialog.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        if (DOM.dialog === undefined) { return false; }
```
Comment (chris-pearce) [inline src/components/dialog/dialog.js:42]: How do we feel about 1 line blocks? Personally I'm not a fan but happy to go with what the majority wants.

File: src/components/dialog/dialog.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+    // Combine defaults with passed in options
+    const settings = Object.assign(defaults, userOptions);
+    
+    init();
+
+    /**
+    * @function init
+    * @desc Initialises the dialog
+    */
```
Comment (chris-pearce) [inline src/components/dialog/dialog.js:29]: Indent the `*`s by 1 space :+1:.

File: src/components/dialog/dialog.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    function hide() {
+        state.isOpen = false;
+        DOM.dialog.classList.add('is-hidden');
+        DOM.backdrop.classList.add('is-hidden');
+        DOM.page.removeAttribute('data-current-dialog');
+    };
+    
+    /**
+     * @function keyHandler
+     * @desc Checks to see if escape (key 27) has been pressed and dialog not modal
```
Comment (chris-pearce) [inline src/components/dialog/dialog.js:81]: Do we need to the _and dialog not modal_ check on `closeBtn` and `backdrop` to?

**Comment by chris-pearce** [comment]
Looking good @bnhovde :+1:. There's more functionality around accessibility but not sure if you plan to get to that later? If you need a hand with any of it I can provide you with the specs. Cheers.

File: src/components/dialog/dialog.js (hunk 1 above)
Comment (bnhovde) [inline src/components/dialog/dialog.js:18]: Yeah like it. We need a style guide don't we? Hehe..

File: src/components/dialog/dialog.js (hunk 2 above)
Comment (bnhovde) [inline src/components/dialog/dialog.js:10]: Old habits mate! Good spot

File: src/components/dialog/dialog.js (hunk 4 above)
Comment (bnhovde) [inline src/components/dialog/dialog.js:42]: I'll also go with the majority.

My personal preference is towards single-line for single statements, for both js and css.

For example, when a map function only performs one operation:

`[1, 2, 3].map(function (num) { return num * 2 })`

but with a second operation comes wrappage:

```
[1, 2, 3, 4].map(num => {
    let multiplier = 2 + num
    return num * multiplier
})
```

Anyway, this is one of those cases where there's no real rational behind it other than I think it looks nicer, so let democracy decide!

Another one for the style guide.

**Comment by bnhovde** [comment]
@chris-pearce Yeah I think It'll need a round in the accessibility ring!

I'm just merging this in now, forgot it was still pending.

**Pull request merged**